src/OMSimulatorLib/MatVer4.cpp

```diff
@@ -119,6 +122,11 @@ oms::MatVer4Matrix* oms::readMatVer4Matrix(FILE* file)
   MatVer4Type_t type = (MatVer4Type_t) (matrix->header.type % 100);
   size_t size = sizeofMatVer4Type(type);
   matrix->data = malloc(matrix->header.mrows * matrix->header.ncols * size);
+  if (!matrix->data)
+  {
+    printf("Out of memory");
+    return NULL;
+  }
```

Comment:
This shouldn't print any error, but the function calling it should check for the return value and issue a warning. But I guess this is just for testing right now.

Reply:
But `matrix->data` is used in the next line to read data into it. If we are out of memory this should be a problem.

Reply:
Yes, you should return, but not print from inside the function.

Reply:
Ah, I see.